uniswap-v2-core/test/UniswapV2Pair.spec.ts

```diff
@@ -437,4 +439,62 @@ describe('UniswapV2Pair', () => {
     expect(await token0.balanceOf(pair.address), "Token 0 balance of pair").to.eq(bigNumberify(1000).add(token0ExpBalVexchange))
     expect(await token1.balanceOf(pair.address), "Token 1 balance of pair").to.eq(bigNumberify(1000).add(token1ExpBalVexchange))
   })
+
+
+  /**
+   *  recoverToken - error handling for invalid tokens 
+   */
+  it('recoverToken:invalidToken', async () => {
+    await expect(pair.recoverToken(token0.address)).to.be.revertedWith('Vexchange: INVALID_TOKEN_TO_RECOVER')
+    await expect(pair.recoverToken(token1.address)).to.be.revertedWith('Vexchange: INVALID_TOKEN_TO_RECOVER')
```

Comment:
This is fine, would break if order of `require()` calls changed but the test case is sufficient for what we need

Reply:
you mean if we had swapped these two lines in `UniswapV2Pair.sol`, the test case wouldn't work? 

` require(token != token0, "Vexchange: INVALID_TOKEN_TO_RECOVER"); `
`require(token != token1, "Vexchange: INVALID_TOKEN_TO_RECOVER");` 

If I'm not mistaken, I think it still would

Reply:
```
require(recoverer != address(0), "Vexchange: RECOVERER_ZERO_ADDRESS");
```

Would trigger on this test case if it was the first require?

Reply:
you're right, it would. So do you think I should craft the test case such that the pair has a recoverer address, so that we are only testing for the validity of the token address?

Reply:
Yes, while a bit pedantic, it is in-line with the goal of testing a single thing at a time